p2p/peer.go

```diff
@@ -490,35 +761,76 @@ func (s *peerStore) List() ([]*peerInfo, error) {
 	return peers, nil
 }
 
-// peerInfo contains peer information stored in a peerStore.
+// Ranked returns a list of peers ordered by score (better peers first).
+// Peers with equal scores are returned in an arbitrary order.
```

Comment:
We can sort by NodeID in case of ties for deterministic reasoning.

Reply:
That's what I initially did, but e.g. in a 100-node network with `MaxConnected=20` then all nodes in the network would only connect to the same 20 peers, which would be bad.

Reply:
I see. So we kind of want the pseudo-randomness?

Reply:
I was thinking a little about this. Is it an ideal characteristic that nodes avoid staying with the same peers even if they have a high rank and have some small element of exploration?

Reply:
Great, we can add this to #5670 and resolve this conversation.